CMakeLists.txt

```diff
@@ -161,8 +161,12 @@ endif()
 if(LINK_WITH_PYTHON)
   # find_package(Python) has the problem that on github actions in particular it'll pick up the most recent python (eg 3.9) from the tool cache
   # even if you have used the setup-python action and set it to 3.8, so we make the exact version required
-  if(PYTHON_VERSION)
-    find_package(Python ${PYTHON_VERSION} EXACT COMPONENTS Interpreter Development REQUIRED)
+  set(Python_REQUIRED_VERSION "" CACHE STRING "The specific Python version you want to link to (Optional)")
+  set(Python_ROOT_DIR "" CACHE PATH "The path to the root installation of your python version, if you want to explicitly set it (Optional)")
+  mark_as_advanced(Python_REQUIRED_VERSION)
+  mark_as_advanced(Python_ROOT_DIR)
+  if(Python_REQUIRED_VERSION)
+    find_package(Python ${Python_REQUIRED_VERSION} EXACT COMPONENTS Interpreter Development REQUIRED)
   else()
     find_package(Python 3.6 COMPONENTS Interpreter Development REQUIRED)
```

Comment:
My only comment is that this should probably be 3.7 and not 3.6 now since Python 3.6 reached End of Life Dec 2021. But this doesn't have to change in this PR necessarily.

Reply:
That would be annoying for people on Ubuntu 18 04 (or Gort, until yesterday) as the system python is 3.6

Reply:
@mbadams5 I do hear you there.  Once Ubuntu 22.04 is released next month I plan to reimage our CI machines up to 20.04.  That would still be one full LTS behind, which seems pretty safe.  In that case, I'd be fully in support of going to 3.7 here.  Let's chat about that soon.